documentation/rules/NS5001.md

```diff
@@ -0,0 +1,59 @@
+﻿# NS5001
```

Comment:
@dtchepak as discussed https://github.com/nsubstitute/NSubstitute/issues/604#issuecomment-584941450
feel free to suggest better documentation and diagnostic id